datafusion/physical-expr-common/src/aggregate/groups_accumulator/accumulate.rs

```diff
@@ -324,6 +324,18 @@ impl NullState {
         }
     }
 
+    /// Check if the accumulated value for the group at the given `index` is valid,
+    /// meaning that there was at least one value passing the filter for this group.
+    pub fn is_valid(&self, index: usize) -> bool {
+        self.seen_values.get_bit(index)
+    }
```

Comment:
Ahh I can't use it in `accumulate` due to mutable vs immutable borrow. This API is pretty difficult to use.

Reply:
I don't understand this comment 🤔

Reply:
This is the signature of `accumulate`:
```rust
    pub fn accumulate<T, F>(
        &mut self,
        group_indices: &[usize],
        values: &PrimitiveArray<T>,
        opt_filter: Option<&BooleanArray>,
        total_num_groups: usize,
        mut value_fn: F,
    ) where
        T: ArrowPrimitiveType + Send,
        F: FnMut(usize, bool, T::Native) + Send
```

But I can't use `is_null` inside `value_fn` because the `NullState` is already borrowed as mutable

Reply:
Ideally `BooleanBufferBuilder::set_bit` should return the previous value and we can pass it along to the callback. That's in arrow I guess.